### PR TITLE
add flexible timeout for headless crawl

### PR DESCRIPTION
**Pull request by thr3athunt3r**
Hard-coded timeouts can sometimes raise exceptions.

### CODE REVIEW

**Comment by bretfourbe** [comment]
Hi @thr3athunt3r, can you create an issue to explain the problem and give examples and what behavior do you expect from wapiti ?
Here it seems that timeout is still hard-coded, you just changed the value of timeout.

File: wapitiCore/controller/wapiti.py [hunk 1]
```diff
@@ -187,6 +186,7 @@ def __init__(self, scope_request: Request, scope="folder", session_dir=None, con
         self.detailed_report = False
         self._headless_mode = "no"
         self._wait_time = 2.
+        self._timeout = 6.
```
Comment (devl00p) [inline wapitiCore/controller/wapiti.py:189]: `_timeout` should not be declared in the `Wapiti` class as a timeout is already stored in the `CrawlerConfiguration` object.

File: wapitiCore/net/intercepting_explorer.py [hunk 2]
```diff
@@ -379,7 +379,8 @@ def __init__(
             drop_cookies: bool = False,
             headless: str = "no",
             cookies: Optional[CookieJar] = None,
-            wait_time: float = 2.
+            wait_time: float = 2.,
+            timeout: float = 6.,
```
Comment (devl00p) [inline wapitiCore/net/intercepting_explorer.py:383]: No need for this argument as you are using `crawler_configuration.timeout`

Note that you should be able to directly get the timeout using `self._crawler.timeout`

File: wapitiCore/net/intercepting_explorer.py [hunk 3]
```diff
@@ -319,14 +319,14 @@ async def launch_headless_explorer(
 
                 if request.method == "GET":
                     try:
-                        await headless_client.get(request.url_with_fragment, timeout=5)
+                        await headless_client.get(request.url_with_fragment, timeout=timeout)
```
Comment (devl00p) [inline wapitiCore/net/intercepting_explorer.py:322]: I think it can be simplified again using `crawler.timeout`.

**Comment by bretfourbe** [comment]
Hi @thr3athunt3r,
I think this MR fix your issue: https://github.com/wapiti-scanner/wapiti/pull/422 ?

**Pull request closed without merging**